### ISSUE TITLE
style.set_precision(0) displays spurious .0

### ISSUE THREAD

**Issue by xflr6** [pydata/pandas#12134]
The expected display format would be as in [2]:
![screenshot](https://cloud.githubusercontent.com/assets/6342379/12556293/77593db2-c385-11e5-89e1-69a2bdd0a049.png)

**Comment by jreback** [comment]
looks a bit buggy, want to do a PR?

**Comment by xflr6** [comment]
Okay, I guess the [Jinja2 docs](http://jinja.pocoo.org/docs/dev/templates/#round) recommend to do this by changing [this](https://github.com/pydata/pandas/blob/a783d1d4f8d129087c806d326a141b6052bed359/pandas/core/style.py#L120-L122) to something like this:
```jinja2
{% if c.value is number %}
    {% if precision %}
        {{c.value|round(precision)}}
     {% else %}
         {{c.value|round|int}}
     {% endif %}
{% else %}
```
Okay to put another branch like this into the inner loop?

**Comment by jreback** [comment]
I think that is ok, the rendering time is not usually an issue (unless you are rendering millions of lines, which itself is an issue anyhow).

@TomAugspurger

**Comment by xflr6** [comment]
`{{c.value|round|int}}` fails with `nan` and `inf`, so the submitted PR uses percent-formatting

**Comment by TomAugspurger** [comment]
Sorry, I didn't see this until now. I've got branch that is changing how the display formatting works, which is going to conflict with your PR in #12137.  I believe I've fixed this bug in the change ([here](https://github.com/TomAugspurger/pandas/commit/a929d6d4f611f00466871a1c8e2e159d95191133) if you want to take a look at the relevant bit). I'm going to clean that up and submit sometime this week.
